sweep.hh

```diff
@@ -5,42 +5,130 @@
 #include"accum_type.hh"
 #include"random_element.hh"
 #include"get_staples.hh"
+
+#ifdef _USE_OMP_
+#  include<omp.h>
+#endif
+
 #include<random>
 #include<vector>
 
-template<class URNG, class Group> double sweep(gaugeconfig<Group> &U, URNG &engine,
+/**
+ * go through the entire lattice
+ * do N_hit Metropolis-Updates of every link (accept proposed change with probabilty min(1, exp(-deltaS)))
+ * update only changes one link, not the surrounding links, but for the action, every plaquette including this link is needed
+ * -> sum up the unchanged part in staples, ony have to be calculated once for every link
+ * If the lattice is anisotropic with anisotropy xi, the action weights the temporal (including links in direction 0) 
+ * and spatial links differently, this is implemented in get_staples.hh.
+ * For the update, the nearest neighbour links have to be constant, so parallelization is not trivial.
+ * It is done by first updating all even time slices and then all odd time slices.
+ * The overall acceptance rate, as well as the acceptance rate of the temporal links, are measured and returned.
+ * An update means replacing one link U -> R*U, where R is a random element. 
+ * The acceptance rate can be tuned with delta, which determines the possible regions from which R is drawn.
+ * Is it more efficient to use two of eevery variable fro measuring the rates, or would it be better to use vectors for everything?
+ * */
+
+template<class URNG, class Group> std::vector<double> sweep(gaugeconfig<Group> &U, vector<URNG> engine,
                                                const double delta, 
-                                               const size_t N_hit, const double beta) {
+                                               const size_t N_hit, const double beta,
+                                               const double xi=1.0, bool anisotropic=false) {
 
   std::uniform_real_distribution<double> uniform(0., 1.);
   typedef typename accum_type<Group>::type accum;
   size_t rate = 0;
-  Group R;
-  std::vector<size_t> x = {0, 0, 0, 0};
-  for(x[0] = 0; x[0] < U.getLt(); x[0]++) {
-    for(x[1] = 0; x[1] < U.getLx(); x[1]++) {
-      for(x[2] = 0; x[2] < U.getLy(); x[2]++) {
-        for(x[3] = 0; x[3] < U.getLz(); x[3]++) {
+  size_t rate_time = 0;
+  #ifdef _USE_OMP_
+  int threads = omp_get_max_threads();
+  double * omp_acc = new double[threads];
+  double * omp_acc_time = new double[threads];
+  #pragma omp parallel
```

Comment:
Here, as in https://github.com/urbach/su2/pull/19#discussion_r845299576 , I think it may be possible to use ```omp reduce```